[glfw3] Build error

**Host Environment**
- OS: Windows 10


**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install raylib:wasm32-emscripten

**Failure logs**
PS C:\Projects\Emscripten Test> .\vcpkg\vcpkg.exe install raylib:wasm32-emscripten
Computing installation plan...
The following packages will be built and installed:
  * glfw3[core]:wasm32-emscripten -> 3.3.7
    raylib[core,use-audio]:wasm32-emscripten -> 4.0.0#3
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet wasm32-emscripten...
Restored 0 packages from C:\Users\JoPer\AppData\Local\vcpkg\archives in 163.4 us. Use --debug to see more details.
Installing 1/2 glfw3:wasm32-emscripten...
Building glfw3[core]:wasm32-emscripten...
-- Using community triplet wasm32-emscripten. This triplet configuration is not guaranteed to succeed.-- [COMMUNITY] Loading triplet configuration from: C:\Projects\Emscripten Test\vcpkg\triplets\community\wasm32-emscripten.cmake
-- Using cached glfw-glfw-45ce5ddd197d5c58f50fdd3296a5131c894e5527.tar.gz.
-- Cleaning sources at C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Projects/Emscripten Test/vcpkg/downloads/glfw-glfw-45ce5ddd197d5c58f50fdd3296a5131c894e5527.tar.gz
-- Using source at C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean
-- Found external ninja('1.10.2').
-- Configuring wasm32-emscripten
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: C:/mingw64/bin/ninja.exe -v
    Working Directory: C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/wasm32-emscripten-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Projects\Emscripten Test\vcpkg\buildtrees\glfw3\config-wasm32-emscripten-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:331 (vcpkg_execute_required_process)
  ports/glfw3/portfile.cmake:20 (vcpkg_cmake_configure)
  scripts/ports.cmake:146 (include)


error: building glfw3:wasm32-emscripten failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+glfw3
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[glfw3]+Build+error
Include '[glfw3] Build error' in your bug report title, the following version information in your bug 
description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-05-05-67e17c1782801cf481be9ac0b3765dff3e4bdeb8
    vcpkg-scripts version: 261ca0dd4 2022-05-18 (11 hours ago)

**-------\vcpkg\buildtrees\glfw3\config-wasm32-emscripten-out.log-----------------**
[1/2] cmd /c "cd ../../wasm32-emscripten-dbg && "C:/cmake/bin/cmake.exe" "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Projects/Emscripten Test/vcpkg/packages/glfw3_wasm32-emscripten/debug" "-DGLFW_BUILD_EXAMPLES=OFF" "-DGLFW_BUILD_TESTS=OFF" "-DGLFW_BUILD_DOCS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/mingw64/bin/ninja.exe" "-DCMAKE_SYSTEM_NAME=Emscripten" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake" "-DVCPKG_TARGET_TRIPLET=wasm32-emscripten" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=wasm32" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Projects/Emscripten Test/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Projects/Emscripten Test/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
FAILED: ../../wasm32-emscripten-dbg/CMakeCache.txt 
cmd /c "cd ../../wasm32-emscripten-dbg && "C:/cmake/bin/cmake.exe" "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Projects/Emscripten Test/vcpkg/packages/glfw3_wasm32-emscripten/debug" "-DGLFW_BUILD_EXAMPLES=OFF" "-DGLFW_BUILD_TESTS=OFF" "-DGLFW_BUILD_DOCS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/mingw64/bin/ninja.exe" "-DCMAKE_SYSTEM_NAME=Emscripten" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake" "-DVCPKG_TARGET_TRIPLET=wasm32-emscripten" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=wasm32" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Projects/Emscripten Test/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Projects/Emscripten Test/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Using X11 for window creation
CMake Error at C:/cmake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find X11 (missing: X11_X11_LIB)
Call Stack (most recent call first):
  C:/cmake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  C:/cmake/share/cmake-3.23/Modules/FindX11.cmake:457 (find_package_handle_standard_args)
  C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  CMakeLists.txt:208 (find_package)


-- Configuring incomplete, errors occurred!
See also "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/wasm32-emscripten-dbg/CMakeFiles/CMakeOutput.log".
[2/2] cmd /c "cd .. && "C:/cmake/bin/cmake.exe" "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Projects/Emscripten Test/vcpkg/packages/glfw3_wasm32-emscripten" "-DGLFW_BUILD_EXAMPLES=OFF" "-DGLFW_BUILD_TESTS=OFF" "-DGLFW_BUILD_DOCS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/mingw64/bin/ninja.exe" "-DCMAKE_SYSTEM_NAME=Emscripten" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake" "-DVCPKG_TARGET_TRIPLET=wasm32-emscripten" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=wasm32" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Projects/Emscripten Test/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Projects/Emscripten Test/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/cmake/bin/cmake.exe" "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/src/1c894e5527-3070f6d62b.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Projects/Emscripten Test/vcpkg/packages/glfw3_wasm32-emscripten" "-DGLFW_BUILD_EXAMPLES=OFF" "-DGLFW_BUILD_TESTS=OFF" "-DGLFW_BUILD_DOCS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/mingw64/bin/ninja.exe" "-DCMAKE_SYSTEM_NAME=Emscripten" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake" "-DVCPKG_TARGET_TRIPLET=wasm32-emscripten" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=wasm32" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Projects/Emscripten Test/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Projects/Emscripten Test/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Using X11 for window creation
CMake Error at C:/cmake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find X11 (missing: X11_X11_LIB)
Call Stack (most recent call first):
  C:/cmake/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  C:/cmake/share/cmake-3.23/Modules/FindX11.cmake:457 (find_package_handle_standard_args)
  C:/Projects/Emscripten Test/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  CMakeLists.txt:208 (find_package)


-- Configuring incomplete, errors occurred!
See also "C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/wasm32-emscripten-rel/CMakeFiles/CMakeOutput.log".
ninja: build stopped: subcommand failed.

**-----CMakeOutput.log----**
The target system is: Emscripten - 1 - x86
The host system is: Windows - 10.0.19044 - AMD64
Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD succeeded with the following output:
Change Dir: C:/Projects/Emscripten Test/vcpkg/buildtrees/glfw3/wasm32-emscripten-rel/CMakeFiles/CMakeTmp

Run Build Command(s):C:/mingw64/bin/ninja.exe cmTC_9fb08 && [1/2] Building C object CMakeFiles/cmTC_9fb08.dir/src.c.o

[2/2] Linking C executable cmTC_9fb08.js



Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}



**Additional context**
I have very limited knowledge about bouth emsdk and vcpk as this is the first project I've tried to setup with this configuration. 
I pretty much just followd the steps from this site: https://thatonegamedev.com/cpp/programming-a-c-game-for-the-web-emscripten/
Thank you.

--- a/ports/raylib/portfile.cmake
+++ b/ports/raylib/portfile.cmake
@@ -14,6 +14,10 @@ if(VCPKG_TARGET_IS_OSX OR VCPKG_TARGET_IS_LINUX)
     set(patches fix-linkGlfw.patch)
 endif()
 
+if(VCPKG_TARGET_IS_EMSCRIPTEN)
+	set(ADDITIONAL_OPTIONS "-DPLATFORM=Web")
+endif()
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO raysan5/raylib
@@ -46,6 +50,7 @@ vcpkg_cmake_configure(
         -DSTATIC=${STATIC}
         -DUSE_EXTERNAL_GLFW=OFF # externl glfw3 causes build errors on Windows
         ${FEATURE_OPTIONS}
+        ${ADDITIONAL_OPTIONS}
     OPTIONS_DEBUG
         -DENABLE_ASAN=${DEBUG_ENABLE_SANITIZERS}
         -DENABLE_UBSAN=${DEBUG_ENABLE_SANITIZERS}

--- a/ports/raylib/portfile.cmake
+++ b/ports/raylib/portfile.cmake
@@ -21,8 +21,8 @@ endif()
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO raysan5/raylib
-    REF bf2ad9df5fdcaa385b2a7f66fd85632eeebbadaa #v4.2.0
-    SHA512 f6b1738d96fef89059062f570f67aaa8b143ccfbee78abfe5fbb25083371a4c432f3d1d0d357e4b475b4b72a6db7823c2341b70ac947759603b033c2b0acec47
+    REF 387c06000618ef0aa3b15c5e46d1c525ba194c50 #v4.2.0
+    SHA512 df7640dcf2979128586c7b671da4f17b628bed8a4526f252469d2d25aee2c9b16695087c8766c10bfdf26a89ff1a8f7ff6005baa7c0dbf77dadc8a0c469e1217
     HEAD_REF master
     PATCHES ${patches}
 )

--- a/ports/raylib/vcpkg.json
+++ b/ports/raylib/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "raylib",
   "version-semver": "4.2.0",
+  "port-version": 1,
   "description": "A simple and easy-to-use library to enjoy videogames programming",
   "homepage": "https://github.com/raysan5/raylib",
   "license": "Zlib",

--- a/ports/raylib/vcpkg.json
+++ b/ports/raylib/vcpkg.json
@@ -8,7 +8,7 @@
   "dependencies": [
     {
       "name": "glfw3",
-      "platform": "!windows"
+      "platform": "!(windows | emscripten)"
     },
     {
       "name": "vcpkg-cmake",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6218,7 +6218,7 @@
     },
     "raylib": {
       "baseline": "4.2.0",
-      "port-version": 0
+      "port-version": 1
     },
     "rbdl": {
       "baseline": "3.2.0",

--- a/versions/r-/raylib.json
+++ b/versions/r-/raylib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f414adaa0cc7bb24014fb07c4f16647aa853696e",
+      "version-semver": "4.2.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "84270f09d9ecfed2207ea866910905fdfcf8b229",
       "version-semver": "4.2.0",
       "port-version": 0

--- a/versions/r-/raylib.json
+++ b/versions/r-/raylib.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "f414adaa0cc7bb24014fb07c4f16647aa853696e",
+      "git-tree": "f3416dbe0e282bf0df99b8147aa78b0c4c3c0263",
       "version-semver": "4.2.0",
       "port-version": 1
     },

